### PR TITLE
feat: add "fileResolve" option for deep import resolving

### PR DESCRIPTION
**Pull request by KingSora**
Good day!

I wanted to use the `resolve` option in order to resolve files from the `compose:` import. Unfortunately this option only gives me the possibility to resolve the "first layer" of composes.

Consider the following setup:
```css
// a.css
.a {
  composes: b from "./b.css";
}
```
```css
// b.css
.b {
  composes: c from "./c.css";
}
```
```css
// c.css
.c {
  content: 'c';
}
```

In this setup the `a.css` file is my entry point. The `resolve` function will only be fired for the `composes: b from "./b.css"` statement. The "deeper" import inside `b.css` is completely opaque.

For this usecase I've implemented the `fileResolve` option in a non-breaking manner. This function will be called for all `composes` statements no matter how deep they are. 
- It receives two arguments `file` and `importer`, both of which are strings. The `file` is essentially the part between the brackets after the `from` keyword (same as the first argument of the `resolve` option). The `importer` is the file from which this `compose` statements comes from.
- It has to return `string | null` or a `Promise<string | null>`. The returned `string` has to be an absolute path. If `null` is returned the default resultion takes place.

I haven't done that, but with this change I would propose to `deprecate` the current `resolve` option, because the new `fileResolve`
option is a bit more powerful, still straight forward to use and can still do the same thing as the `resolve` function.

The `README` would also need an update, but I would do that after I the feedback from this PR.

Fixes: #131

### CODE REVIEW

**Comment by madyankin** [comment]
@KingSora thanks for the PR, I really appreciate your contribution. I have no time to implement features myself now, so this is a huge help for me.

I’ll review the PR in a few days.

**Comment by KingSora** [comment]
@madyankin looking forward for the review! I've also created another PR which should fix sourcemaps #143

**Comment by KingSora** [comment]
@madyankin just a little reminder :)

**Comment by madyankin** [comment]
@KingSora thanks for pinging me. I have a todo in my Things, but now I'm relocating to another country, so a bit overwhelmed. Will have some time for the review next week

File: src/css-loader-core/loader.js
Comment (madyankin) [inline src/css-loader-core/loader.js:71]: Let's rewrite this using async/await. I'm thinking of rewriting all the promises into async/await in the future

File: src/css-loader-core/loader.js
Comment (madyankin) [inline src/css-loader-core/loader.js:80]: Let's define the variables using separate `let` statements to make them more explicit

File: index.d.ts
Comment (madyankin) [inline index.d.ts:54]: 👍 for updating the typings

File: src/index.js
Comment (madyankin) [inline src/index.js:88]: Thank you for thinking about the backwards compatibility. Anyway, I'd prefer to introduce the breaking change and bump the major version. So, let's remove the original `resolve` version, replace it with yours, and update the readme. Feel free to use the `resolve` option name for this.

**Review by madyankin — CHANGES_REQUESTED** [review]
@KingSora hey, I found some time to review the PR. I hope you're not demotivated by such a long time waiting for the review. Could you please make a few changes in the PR?

**Comment by KingSora** [comment]
@madyankin please leave it open. I'll come back to it as soon as possible (next week possibly) I just need to finish an different project first

**Comment by madyankin** [comment]
@KingSora don't worry — I have merged it and fixed the things I asked. Thank you for the implementation!

I'm going to release the new version after merging #143 — there are some failing tests. Could you take a look please?

**Comment by KingSora** [comment]
@madyankin on it

**Pull request closed without merging**
